app/Services/Pocket.php

```diff
@@ -0,0 +1,221 @@
+<?php namespace App\Services;
+
+use \App\Items as ItemModel;
+use \App\UserItems as UserItemsModel;
+use GuzzleHttp\Exception\ClientException;
+use Goose\Client as GooseClient;
+use Readability;
+use \Log;
+
+class Pocket {
+
+    const POCKET_API_URL = "https://getpocket.com/v3/";
+
+    /**
+     * User to make pocket requests on behalf of.
+     * @var
+     */
+    protected $user;
+
+    /**
+     * @var \GuzzleHttp\Client $httpClient
+     */
+    protected $httpClient;
+
+    public function __construct($user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Add latest pock
+     *
+     * @throws \Exception
+     */
+    public function synchronisePocketItems()
+    {
+
+        $feed = $this->getPocketFeed();
+
+        $urls = array();
+        foreach ($feed as $feedItem) {
+            $item = $this->addItem($feedItem['resolved_url'], $feedItem['resolved_title'], $feedItem['excerpt']);
+            if ($item) {
+                $this->addUserItem($item);
+                $urls[] = $feedItem['resolved_url'];
+            }
+        }
+
+        // Remove previous items which are no longer in the feed
+        $affectedRows = UserItemsModel::whereNotIn('item_id', array_map("md5", $urls))->delete();
```

Comment:
There's no condition on the user_id here! Does this mean we're removing all items except for the ones that appear in the current user's feed? For each user?